### PR TITLE
Support linear rope scale

### PR DESCRIPTION
**Pull request by lvhan028**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

- [ ] profiling

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -374,7 +373,7 @@ add_library(transformer-shared SHARED
 
 if (BUILD_MULTI_GPU)
 target_link_libraries(transformer-shared PUBLIC
-  -lmpi
+  ${MPI_CXX_LIBRARIES}
```
Comment (lzhangzz) [inline CMakeLists.txt:376]: How is this related to this PR?

File: src/turbomind/utils/CMakeLists.txt [hunk 2]
```diff
@@ -63,7 +63,7 @@ add_library(mpi_utils STATIC mpi_utils.cc)
 set_property(TARGET mpi_utils PROPERTY POSITION_INDEPENDENT_CODE  ON)
 set_property(TARGET mpi_utils PROPERTY CUDA_RESOLVE_DEVICE_SYMBOLS  ON)
 if (BUILD_MULTI_GPU)
-    target_link_libraries(mpi_utils PUBLIC mpi logger)
+    target_link_libraries(mpi_utils PUBLIC ${MPI_CXX_LIBRARIES} logger)
```
Comment (lzhangzz) [inline src/turbomind/utils/CMakeLists.txt:66]: How is this related to this PR?

File: CMakeLists.txt (hunk 1 above)
Comment (lvhan028) [reply CMakeLists.txt:376]: Not related. I'll remove it and pull another PR

File: src/turbomind/utils/CMakeLists.txt (hunk 2 above)
Comment (lvhan028) [reply src/turbomind/utils/CMakeLists.txt:66]: Not related. I'll remove it and pull another PR

**Comment by lvhan028** [comment]
`turbomind-2.1` branch has supported this feature.
